### PR TITLE
fix(button): apply color classes correctly.

### LINKED ISSUE #75: md-button attaches class "md-undefined" to host element when on color provided
Given an html snippet with an md-button directive like so:

```html
<button md-button>flat</button>
```

This produces the following ouput:

```html
<button md-button ... class="md-undefined">...</button>
```

What it should produce is:

```html
<button md-button ...>...</button>
```

The `md-undefined` class is getting attached due to [this code](https://github.com/angular/material2/blob/master/src/components/button/button.ts#L21-L40) since `setClassList()` returns `"md-undefined"` when the `color` attribute is unset. 

I tried working around the by returning an empty string when `color` is undefined, but that resulted in this html:

```html
<button md-button ... class>...</button>
```

Which is also invalid.

### PR DESCRIPTION
**Pull request by DevVersion**
Fixes #75. Closes #89

### CODE REVIEW

File: src/components/button/button.ts [hunk 1]
```diff
@@ -48,6 +57,18 @@ export class MdButton {
     TimerWrapper.setTimeout(() => { this.isMouseDown = false; }, 100);
   }
 
+  _updateColor(newColor: string) {
+    if (this._color != null && this._color != '') {
+      this.renderer.setElementClass(this.elementRef.nativeElement, `md-${newColor}`, false);
```
Comment (jelbourn) [inline src/components/button/button.ts]: Should be `this._color` in the template expression

Comment (DevVersion) [inline src/components/button/button.ts]: Indeed -  fixed it, thanks!

File: src/components/button/button.ts [hunk 2]
```diff
@@ -48,6 +57,18 @@ export class MdButton {
...
   }
 
+  _updateColor(newColor: string) {
+    if (this._color != null && this._color != '') {
+      this.renderer.setElementClass(this.elementRef.nativeElement, `md-${this._color}`, false);
+    }
+
+    if (newColor != null && newColor != '') {
+      this.renderer.setElementClass(this.elementRef.nativeElement, `md-${newColor}`, true);
+    }
```
Comment (kara) [inline src/components/button/button.ts]: There's some repetition between lines 61-63 and lines 65-67. Can you refactor to share more code, like:

```javascript
set color(value: string) {
   this._updateColor(this._color, false);
   this._updateColor(value, true);
}

_updateColor(color: string, bool: boolean): void {
   if (color != null && color != '') {
      this.renderer.setElementClass(this.elementRef.nativeElement, `md-${color}`, bool);
   }
}
```

Comment (DevVersion) [inline src/components/button/button.ts]: @kara Changed it as discussed. Thanks for the suggestion.

**Comment by jelbourn** [comment]
LGTM

**Pull request closed without merging**